pkg/aws/apis/validation/validation.go

```diff
@@ -74,52 +76,92 @@ func validateSpecTags(tags map[string]string) []error {
 	}
 
 	if clusterName == "" {
-		allErrs = append(allErrs, fmt.Errorf("Tag is required of the form kubernetes.io/cluster/****"))
+		allErrs = append(allErrs, field.Required(fldPath.Child(""), "Tag required of the form kubernetes.io/cluster/****"))
 	}
 	if nodeRole == "" {
-		allErrs = append(allErrs, fmt.Errorf("Tag is required of the form kubernetes.io/role/****"))
+		allErrs = append(allErrs, field.Required(fldPath.Child(""), "Tag required of the form kubernetes.io/role/****"))
 	}
+
 	return allErrs
 }
 
-func validateBlockDevices(blockDevices []awsapi.AWSBlockDeviceMappingSpec) []error {
+func validateBlockDevices(blockDevices []awsapi.AWSBlockDeviceMappingSpec, fldPath *field.Path) field.ErrorList {
+	var allErrs field.ErrorList
+
+	rootPartitionCount := 0
+	var deviceNames = make(map[string]int)
+
+	validVolumeTypes := []string{"gp2", "gp3", "io1", "st1", "sc1", "standard"}
+
+	// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/device_naming.html
+	const dataDeviceNameFmt string = `/dev/(sd[a-z]|xvd[a-c][a-z]?)`
+	var dataDeviceNameRegexp = regexp.MustCompile("^" + dataDeviceNameFmt + "$")
 
-	var allErrs []error
+	// if blockDevices is empty, AWS will automatically create a root partition
+	for i, disk := range blockDevices {
+		idxPath := fldPath.Index(i)
 
-	if len(blockDevices) > 1 {
-		allErrs = append(allErrs, fmt.Errorf("Can only specify one (root) block device"))
-	} else if len(blockDevices) == 1 {
-		if blockDevices[0].Ebs.VolumeSize <= 0 {
-			allErrs = append(allErrs, fmt.Errorf("Please mention a valid ebs volume size"))
+		if disk.DeviceName == "/root" {
+			rootPartitionCount++
+		} else if len(blockDevices) > 1 && !dataDeviceNameRegexp.MatchString(disk.DeviceName) {
+			// if there are multiple devices, non-root devices are expected to adhere to AWS naming conventions
+			allErrs = append(allErrs, field.Invalid(idxPath.Child("deviceName"), disk.DeviceName, utilvalidation.RegexError(fmt.Sprintf("Device name given: %s does not match the expected pattern", disk.DeviceName), dataDeviceNameFmt)))
 		}
-		if blockDevices[0].Ebs.VolumeType == "" {
-			allErrs = append(allErrs, fmt.Errorf("Please mention a valid ebs volume type"))
-		} else if blockDevices[0].Ebs.VolumeType == "io1" && blockDevices[0].Ebs.Iops <= 0 {
-			allErrs = append(allErrs, fmt.Errorf("Please mention a valid ebs volume iops"))
+
+		if _, keyExist := deviceNames[disk.DeviceName]; keyExist {
+			deviceNames[disk.DeviceName]++
+		} else {
+			deviceNames[disk.DeviceName] = 1
+		}
+
+		if !contains(validVolumeTypes, disk.Ebs.VolumeType) {
+			allErrs = append(allErrs, field.Required(idxPath.Child("ebs.volumeType"), "Please mention a valid EBS volume type: gp2, gp3, io1, st1, sc1, or standard"))
 		}
+
+		if disk.Ebs.VolumeSize <= 0 {
+			allErrs = append(allErrs, field.Required(idxPath.Child("ebs.volumeSize"), "Please mention a valid EBS volume size"))
+		}
+
+		if disk.Ebs.VolumeType == "io1" && disk.Ebs.Iops <= 0 {
```

Comment:
Might be better to move `io1` as a constant [here](https://github.com/gardener/machine-controller-manager-provider-aws/blob/master/pkg/aws/apis/aws_provider_spec.go)